### PR TITLE
DEV: When filter param is present, add param to nav-item hrefs

### PR DESCRIPTION
**Pull request by markvanlan**
This also adds a new function to the JS plugin api: `addTopicTrackingFilterFn` which allows plugins to manipulate topic tracking state count for nav items

### CODE REVIEW

**Comment by markvanlan** [comment]
@eviltrout This commit https://github.com/discourse/discourse/pull/10299/commits/984fa255d5874774a811c202d06e6696c039d51a

Added the `filter` query param in the wrong route. This moves 
```
  queryParams: {
    filter: { refreshModel: true }
  },
```

to the correct route.

File: app/assets/javascripts/discourse/app/models/topic-tracking-state.js [hunk 1]
```diff
@@ -6,6 +6,12 @@ import PreloadStore from "discourse/lib/preload-store";
 import Category from "discourse/models/category";
 import User from "discourse/models/user";
 
+const topicTrackingFilterFns = [];
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/models/topic-tracking-state.js:9]: make sure we don't pile some state between tests

File: app/assets/javascripts/discourse/app/models/nav-item.js [hunk 2]
```diff
@@ -69,6 +74,11 @@ const NavItem = EmberObject.extend({
     return mode + name.replace(" ", "-");
   },
 
+  @discourseComputed("currentRouteQueryParams.filter")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/models/nav-item.js:77]: could use `equal`

https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Fcomputed/equal

File: app/assets/javascripts/discourse/app/models/topic-tracking-state.js (hunk 1 above)
Comment (markvanlan) [reply app/assets/javascripts/discourse/app/models/topic-tracking-state.js:9]: how?

**Comment by SamSaffron** [comment]
lets use https://github.com/discourse/discourse/commit/eb18f91cf70feb0052b3c063283fad833623db73 for tracking state changes.

**Pull request closed without merging**